python/pipeline/utils/performance.py

```diff
@@ -406,12 +406,18 @@ def parallel_motion_stack(chunks, results, raster_phase, fill_fraction,
         if abs(raster_phase) > 1e-7:
             field = galvo_corrections.correct_raster(field, raster_phase, fill_fraction)
 
+        # Compute initial template by averaging 10 frames that correlate highly with middle one
+        num_frames = field.shape[-1]
+        frames = field.reshape((-1, num_frames)) # num_pixels x num_frames
+        residuals = frames - frames.mean(axis=0)
+        frames_std = frames.std(axis=0)
+        covs = np.mean(residuals.T * residuals[:, int(num_frames / 2)], axis=-1)
```

Comment:
Should it be `@` operator instead of `*`? Otherwise the size of covariance matrix will be num_frames * num_pixels, and so is the correlation matrix. I think they should both be vectors.

Reply:
I'm calculating here only one row of the correlation matrix so covs and corr are a single [num_frames] vector: covariance of each i with the middle frame and correlations of each i with the middle frame. The way to get the entire [num_frames x num_frames] correlation matrix will be something like `np.dot(residuals.T, residuals)/(frames.shape[0] * np.outer(frames_std, frames_std))` which looks nicer but is slightly more work. Here residuals.T is [num_frames x num_pixes] and residuals[:, int(num_frames / 2) is [num_pixels], the * product will return a num_frames x num_pixels and I take the mean over all pixels to get the covariances per frame it's a bit dirty but I think it works